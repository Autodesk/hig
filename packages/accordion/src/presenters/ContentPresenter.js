import React, { useRef, useLayoutEffect, useState } from "react";
import PropTypes from "prop-types";
import { Transition } from "react-transition-group";
import { css, cx } from "emotion";
import { createCustomClassNames } from "@weave-design/utils";
import stylesheet from "../stylesheet";

const ContentPresenter = (props) => {
  const { children, collapsed, className } = props;

  const contentContainerRef = useRef(null);

  const [contentOverflow, setContentOverflow] = useState("hidden");
  const [contentHeight, setContentHeight] = useState(0);
  const [isAnimationAllowed, setIsAnimationAllowed] = useState(false);

  useLayoutEffect(() => {
    if (contentContainerRef.current)
      setContentHeight(contentContainerRef.current.scrollHeight);
  }, [children, contentContainerRef]);

  const onAnimationDone = (callback, duration) => {
    setTimeout(() => callback(), duration);
  };

  const { createTransitionStyles } = stylesheet();

  const { transitionStyles, duration, defaultStyle } = createTransitionStyles({
    contentHeight,
    contentOverflow,
    isAnimationAllowed,
  });

  const contentClassName = createCustomClassNames(className, "content");

  return (
    <Transition
      nodeRef={contentContainerRef}
      timeout={duration}
      in={!collapsed}
      onEntering={() => {
        setIsAnimationAllowed(true);
      }}
      onExiting={() => {
        setIsAnimationAllowed(true);
      }}
      onEntered={() => {
        onAnimationDone(() => {
          setContentOverflow("visible");
        }, duration);
      }}
      onExited={() => {
        setContentOverflow("hidden");
      }}
    >
      {(state) => (
        <div
          ref={contentContainerRef}
          className={cx(
            css(defaultStyle),
            css(transitionStyles[state]),
            contentClassName
          )}
        >
          {children}
        </div>
      )}
    </Transition>
  );
};

ContentPresenter.propTypes = {
  children: PropTypes.node.isRequired,
  collapsed: PropTypes.bool,
  className: PropTypes.string,
};

export default ContentPresenter;
