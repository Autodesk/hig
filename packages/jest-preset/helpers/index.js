/* eslint-disable global-require */
module.exports = {
  takeSnapshotsOf: require("./takeSnapshotsOf")
};
