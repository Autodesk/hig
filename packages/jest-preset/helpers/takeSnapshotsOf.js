const React = require("react");
const renderer = require("react-test-renderer");

module.exports = function takeSnapshotsOf(Component, cases, rendererOptions) {
  cases.forEach(({ desc, props }) => {
    it(desc, () => {
      const vDom = React.createElement(Component, props);
      const tree = renderer.create(vDom, rendererOptions).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
};
