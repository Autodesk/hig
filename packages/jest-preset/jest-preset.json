{
  "collectCoverageFrom": ["**/src/**/*.{js,jsx}"],
  "coverageDirectory": "<rootDir>/coverage",
  "coveragePathIgnorePatterns": [
    "__gemini__",
    "__stories__",
    "docs",
    "examples/redux",
    "node_modules",
    "packages/jest-preset",
    "packages/vanilla",
    "src/playground"
  ],
  "coverageThreshold": {
    "global": {
      "branches": 50,
      "functions": 50,
      "lines": 50,
      "statements": 50
    }
  },
  "moduleNameMapper": {
    "\\.(css|scss|svg)$": "@hig/jest-preset/fileMock.js"
  },
  "modulePaths": ["node_modules"],
  "moduleFileExtensions": ["js", "jsx", "json"],
  "setupFiles": ["raf/polyfill"],
  "setupTestFrameworkScriptFile": "@hig/jest-preset/setupTests.js",
  "testPathIgnorePatterns": [
    "docs",
    "packages/babel-preset",
    "packages/jest-preset",
    "packages/vanilla"
  ],
  "unmockedModulePathPatterns": ["node_modules"]
}
