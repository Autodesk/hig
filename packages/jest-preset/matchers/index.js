require("./toBeConstants");
require("./toImplementHIGInterfaceOf");
require("./toHavePropTypesAndDocgenInfo");
