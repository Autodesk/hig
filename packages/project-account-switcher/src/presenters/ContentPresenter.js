import React, { Component } from "react";
import PropTypes from "prop-types";
import { css } from "emotion";
import memoize from "lodash.memoize";
import TextLink from "@hig/text-link";
import Typography from "@hig/typography";
import { createButtonEventHandlers } from "@hig/utils";

import constructPlaceholder from "./constructPlaceholder";
import stylesheet from "./stylesheet";

function getTypographyPlaceholderStyles() {
  return {
    textAlign: `center`,
    lineHeight: `32px`
  };
}

export default class ContentPresenter extends Component {
  static propTypes = {
    /** Heading title for the list of Accounts */
    accountTitle: PropTypes.string,
    /** List of Accounts */
    accounts: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    /** Currently selected Account */
    activeAccountObj: PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      label: PropTypes.string
    }),
    /** Currently selected Project */
    activeProjectObj: PropTypes.shape({
      id: PropTypes.string,
      image: PropTypes.string,
      label: PropTypes.string
    }),
    /** Handles Account selection, passed from Behavior */
    onAccountClick: PropTypes.func,
    /** Handles Project selection, passed from Behavior */
    onProjectClick: PropTypes.func,
    /** List of Projects */
    projects: PropTypes.arrayOf(
      PropTypes.shape({
        id: PropTypes.string,
        image: PropTypes.string,
        label: PropTypes.string
      })
    ),
    /** Heading title for the list of Projects */
    projectTitle: PropTypes.string
  };

  static defaultProps = {
    accountTitle: "Accounts",
    projectTitle: "Projects"
  };

  createAccountListItemHandlers = memoize(id =>
    createButtonEventHandlers(event => this.props.onAccountClick(event, id))
  );

  createProjectListItemHandlers = memoize(id =>
    createButtonEventHandlers(event => this.props.onProjectClick(event, id))
  );

  renderAccountsList() {
    const styles = stylesheet();

    return this.props.accounts.map(({ id, image, label }) => {
      const { handleClick, handleKeyDown } = this.createAccountListItemHandlers(
        id
      );

      /** @todo Move this block into it's own component & module */
      return (
        <li
          className={css(styles.switcherItem)}
          key={`account-${id}`}
          onClick={handleClick}
          onKeyDown={handleKeyDown}
          role="menuitem"
          tabIndex="0"
        >
          <span className={css(styles.switcherAccountImageWrapper)}>
            {Boolean(image) && (
              <img className={css(styles.image)} src={image} alt={label} />
            )}
            <Typography
              elementType="span"
              style={getTypographyPlaceholderStyles()}
            >
              {constructPlaceholder(label)}
            </Typography>
          </span>
          <TextLink>{label}</TextLink>
        </li>
      );
    });
  }

  renderProjectsList() {
    const styles = stylesheet();

    return this.props.projects.map(({ id, image, label }) => {
      const { handleClick, handleKeyDown } = this.createProjectListItemHandlers(
        id
      );

      /** @todo Move this block into it's own component & module */
      return (
        <li
          key={`project-${id}`}
          className={css(styles.switcherItem)}
          role="menuitem"
          tabIndex="0"
          onClick={handleClick}
          onKeyDown={handleKeyDown}
        >
          <span className={css(styles.imageWrapper)}>
            {Boolean(image) && (
              <img className={css(styles.image)} src={image} alt={label} />
            )}
            <Typography
              elementType="span"
              style={getTypographyPlaceholderStyles()}
            >
              {constructPlaceholder(label)}
            </Typography>
          </span>
          <TextLink>{label}</TextLink>
        </li>
      );
    });
  }

  render() {
    const { accounts, accountTitle, projects, projectTitle } = this.props;
    const styles = stylesheet();
    const typographyTitleStyles = {
      textTransform: `uppercase`,
      fontSize: `11px`
    };

    return (
      <div>
        {accounts && (
          <ul className={css(styles.switcherList)}>
            <Typography elementType="span" style={typographyTitleStyles}>
              {accountTitle}
            </Typography>
            {this.renderAccountsList()}
          </ul>
        )}
        {projects && (
          <ul className={css(styles.switcherList)}>
            <Typography elementType="span" style={typographyTitleStyles}>
              {projectTitle}
            </Typography>
            {this.renderProjectsList()}
          </ul>
        )}
      </div>
    );
  }
}
