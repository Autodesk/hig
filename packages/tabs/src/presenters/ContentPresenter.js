import React from "react";
import { cx, css } from "emotion";
import PropTypes from "prop-types";
import { createCustomClassNames } from "@hig/utils";
import stylesheet from "./Tab.stylesheet";

export default function ContentPresenter({
  children,
  stylesheet: customStylesheet,
  ...otherProps
}) {
  const { className } = otherProps;
  const styles = stylesheet({ stylesheet: customStylesheet }, {});
  const customClassName = createCustomClassNames(className, "content");
  return (
    <div className={cx(css(styles.content), customClassName)}>{children}</div>
  );
}

ContentPresenter.propTypes = {
  children: PropTypes.node,
  stylesheet: PropTypes.func
};
