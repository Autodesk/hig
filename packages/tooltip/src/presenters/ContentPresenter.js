import React from "react";
import PropTypes from "prop-types";
import { css } from "emotion";

import stylesheet from "./stylesheet";

export default function ContentPresenter({ children }) {
  const styles = stylesheet();

  return <div className={css(styles.content)}>{children}</div>;
}

ContentPresenter.propTypes = {
  /** Content */
  children: PropTypes.node
};
