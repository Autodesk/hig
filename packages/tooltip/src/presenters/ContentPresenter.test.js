import React from "react";
import { takeSnapshotsOf } from "@hig/jest-preset/helpers";

import ContentPresenter from "./ContentPresenter";

describe("tooltip/presenters/ContentPresenter", () => {
  takeSnapshotsOf(ContentPresenter, [
    {
      description: "renders without props",
      props: {}
    },
    {
      description: "renders with children",
      props: {
        children: <span>Hello</span>
      }
    }
  ]);
});
