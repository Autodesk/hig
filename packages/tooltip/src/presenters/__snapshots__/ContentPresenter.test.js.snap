// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip/presenters/ContentPresenter renders with children 1`] = `
.emotion-0 {
  min-height: 32px;
  padding: 8px 12px;
  box-sizing: border-box;
}

<div
  className="emotion-0"
>
  <span>
    Hello
  </span>
</div>
`;

exports[`tooltip/presenters/ContentPresenter renders without props 1`] = `
.emotion-0 {
  min-height: 32px;
  padding: 8px 12px;
  box-sizing: border-box;
}

<div
  className="emotion-0"
/>
`;
